projects/train/train/data/waveforms/generator/cbc.py

```diff
@@ -103,8 +103,13 @@ def time_domain_strain(self, **parameters):
                 the right of the waveform in seconds
         """
 
-        # TODO: support time domain waveforms
-        hc, hp = self.waveform(self.frequencies[self.freq_mask], **parameters)
+        device = parameters["chirp_mass"].device
+        freqs = torch.clone(self.frequencies).to(device)
```

Comment:
What's the reason for cloning the frequency grid here? Does passing it into the approximant change it?

Reply:
The `self.frequency` can be on the cpu. So I clone it and transfer to the gpu for use in this function, without impacting the global attribute.